src/d/actor/d_a_obj_ebomzo.cpp

```diff
@@ -4,118 +4,299 @@
+    /* >> 0x1E mSpl*/ 0,
+    /* >> 0x1F mMtrl*/ 0,
+    /* >> 0x20 mBase*/ {
+        /* >>> 0x20 mSPrm*/ 0
+    }
+},
+
+/* > 0x24 mGObjTg*/ {
+    /* >> 0x24 mSe*/ 0,
+    /* >> 0x25 mHitMark*/ 0,
+    /* >> 0x26 mSpl*/ 0,
+    /* >> 0x27 mMtrl*/ 0,
+    /* >> 0x28 mBase*/ {
+        /* >>> 0x28 mSPrm*/ 0
+    }
+},
+
+/* > 0x2C mGObjCo*/ {
+    /* >> 0x2C mBase*/ {
+        /* >>> 0x2C mSPrm*/ 0
+     }
+},
+
+},
+
+/*0x30 mSphAttr*/{
+    /* > 0x30 mSph*/{
+        /* >> 0x30 mCenter*/ {0,0,0},
+        /* >> 0x3C mRadius*/ 10.0f
+    }
+}
+};
 
 /* 00000078-0000012C       .text CreateHeap__Q211daObjEbomzo5Act_cFv */
-void daObjEbomzo::Act_c::CreateHeap() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, 4);
+    JUT_ASSERT(140, model_data != 0);
+    mpModel = mDoExt_J3DModel__create(model_data, 0, 0x11020203);
+    return mpModel != NULL;
 }
 
 /* 0000012C-0000021C       .text Create__Q211daObjEbomzo5Act_cFv */
-s32 daObjEbomzo::Act_c::Create() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Create() {
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    init_mtx();
+
+    fopAcM_setCullSizeBox(this, -1000.0, -100.0, -1000.0, 1000.0, 1000.0, 1000.0);
+
+    mStts.Init(0xff, 0xff, this);
+    mCollider.Set(sph_check_src);
+    mCollider.SetStts(&mStts);
+
+    mXRotRate = 0;
+    mpParticleEmitter = NULL;
+
+    if (fopAcM_isSwitch(this, prm_get_swSave())) {
+        mMode = Ebomzo_Mode_Fall;
+        current.angle.x = 0x4000;
+    }
+    else mMode = Ebomzo_Mode_Check;
+
+  return TRUE;
 }
 
 /* 0000021C-000003A0       .text Mthd_Create__Q211daObjEbomzo5Act_cFv */
-void daObjEbomzo::Act_c::Mthd_Create() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Mthd_Create() {
+    int phase_state;
+
+    fopAcM_SetupActor(this, daObjEbomzo::Act_c);
+    
+    phase_state = dComIfG_resLoad(&mPhs, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, 7, NULL, 0x1200);
+        JUT_ASSERT(194, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+    }
+    return phase_state;
 }
 
 /* 00000558-00000560       .text Delete__Q211daObjEbomzo5Act_cFv */
-BOOL daObjEbomzo::Act_c::Delete() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Delete() {
+    return TRUE;
 }
 
 /* 00000560-000005AC       .text Mthd_Delete__Q211daObjEbomzo5Act_cFv */
-void daObjEbomzo::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Mthd_Delete() {
+    u32 result = MoveBGDelete();
+    dComIfG_resDelete(&mPhs, M_arcname);
+    return result;
 }
 
 /* 000005AC-0000062C       .text set_mtx__Q211daObjEbomzo5Act_cFv */
 void daObjEbomzo::Act_c::set_mtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+
+    PSMTXCopy(mDoMtx_stack_c::now, mpModel->mBaseTransformMtx);
+    PSMTXCopy(mDoMtx_stack_c::now, M_tmp_mtx);
```

Comment:
```suggestion
    MTXCopy(mDoMtx_stack_c::now, mpModel->mBaseTransformMtx);
    MTXCopy(mDoMtx_stack_c::now, M_tmp_mtx);
```

Reply:
mpModel->setBaseTRMtx() and cMtx_copy, no?

Reply:
ah yeah, you're right